src/libsync/owncloudpropagator.cpp

```diff
@@ -608,22 +610,23 @@ bool PropagateDirectory::scheduleNextJob()
 
     bool stopAtDirectory = false;
     // FIXME: use the cached value of finished job
```

Comment:
Here is what you need to do ro remove the O(N^2)   cache the value of the first unfinished job and start from here instead of always starting from 0